`DateTimePicker`: fix date format when switching to 12-hour time format (#37465)
WordPress/gutenberg: packages/components/src/date-time/README.md

### is12Hour

> Whether we use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed and the time format is assumed to be MM-DD-YYYY. If its value is `false`, format DD-MM-YYYY will be used.

Reviewer: Nit:

```suggestion
Whether we use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed and the time format is assumed to be `MM-DD-YYYY` (as opposed to the default format `DD-MM-YYYY`).
```
